src/icalendar/tests/test_recurrence.py

```diff
@@ -1,61 +1,42 @@
-from icalendar.caselessdict import CaselessDict
-import unittest
+from datetime import datetime
+
+import pytest
+
+def test_recurrence_properly_parsed(events):
+    assert events.event_with_recurrence['rrule'] == {'COUNT': [100], 'FREQ': ['DAILY']}
+
+@pytest.mark.parametrize('i, exception_date', [
+    (0, datetime(1996, 4, 2, 1, 0)),
+    (1, datetime(1996, 4, 3, 1, 0)),
+    (2, datetime(1996, 4, 4, 1, 0))
+])
+def test_exdate_properly_parsed(events, i, exception_date, in_timezone):
+    assert events.event_with_recurrence['exdate'].dts[i].dt == in_timezone(exception_date, 'UTC')
+
+def test_exdate_properly_marshalled(events):
+    actual = events.event_with_recurrence['exdate'].to_ical()
+    assert actual == b'19960402T010000Z,19960403T010000Z,19960404T010000Z'
+
+# TODO: DOCUMENT BETTER!
```

Comment:
I think, this can become a docstring - just an idea - as it documents the test case.